src/DataManagement/RecurringUpdater.cs

```diff
@@ -12,114 +11,38 @@ public static bool UpdateRecurring(IEntryManager entryManager)
             foreach (var entry in entryManager.EntryList.GetBy(recurring: true))
             {
                 var differenceInMonths = TimeManager.DifferenceInMonths(laterTime: DateTime.Today, earlierTime: entry.Date);
-                if (differenceInMonths > 0)
-                {
-                    if (!UpdateEntry(entryManager, entry, differenceInMonths))
-                    {
-                        return false;
-                    }
-                }
+                if (differenceInMonths <= 0) continue;
+                entry.Date = entry.Date.AddMonths(differenceInMonths);
+                entryManager.Edit(entry.Id, entry);
             }
             return true;
         }
 
-        private static bool UpdateEntry(IEntryManager entryManager, IEntry entry, int months)
+        public static IEntryList CreateRecurringListWithoutOriginalEntry(IEntry entry, EntryType entryType)
```

Comment:
So this now creates list with all needed entries and adds them then, if I understood corectly?

Reply:
The methods in recurring updater are now utilities only, the UpdateRecurring method updates the entries inside an Entry Manager, but the other two methods just form lists of entries that need to be added when a recurring entry is added or editied. This class won't need to be accessed by the front end